wandb/sdk/wandb_run.py

```diff
@@ -3263,13 +3271,25 @@ def _log_artifact(
             wandb.termwarn(
                 "Artifacts logged anonymously cannot be claimed and expire after 7 days."
             )
+
         if not finalize and distributed_id is None:
             raise TypeError("Must provide distributed_id if artifact is not finalize")
+
         if aliases is not None:
-            if any(invalid in alias for alias in aliases for invalid in ["/", ":"]):
-                raise ValueError(
-                    "Aliases must not contain any of the following characters: /, :"
-                )
+            aliases = validate_aliases(aliases)
+
+        # Check if artifact tags are supported
+        if tags is not None:
+            tags = validate_tags(tags)
+
+        supported_artifact_fields = api.api.server_artifact_introspection()
+        if tags and ("tags" not in supported_artifact_fields):
+            wandb.termwarn(
+                "Server does not support Artifact tags, "
+                "please upgrade the server to use Artifact tags."
+            )
+            tags = None
```

Comment:
why is all of this happening the `wandb_run` seems like it shouldn't be done here

Reply:
It seemed more transparent here (not as deep into the function/method call stack).

Also seemed consistent with having other early validation logic here (eg on aliases), and easier to ensure we were covering both core + non-core codepaths.  Definitely open to alternatives that might work better

Reply:
i think it complicates the logic and requires accessing into using internal_api to do graphql queries, ideally, the `wandb_run.Run` functions using our interface that in turn would do all the validation and networking operations, this way we have a clear separation between the user process logic and internal process logic.

Reply:
@kptkin am digging into this more and I'm thinking it may not be a minor change IIUC.  Would we definitely need to absorb the introspection request into the golang code?  If yes, I'm a less sure this'll be small, since
- this would be adding new (instead of repurposing existing) methods/gen code
- we still have to handle non-core mode
- it's trickier to handle errors gracefully from the go client (e.g. IIRC the caught error in the python code has just the str message arg, so I think we'd need to regex-parse and reformat it into a useful warning/error for the user -- let me know if mistaken)

Would you consider this change blocking, and/or are there other alternatives I might have missed?  I lean towards leaving this unchanged, since
- it catches the server incompatibility (+ nulls out tags) early, even if it's an extra request outside the central interface.  so I see the tradeoff, but it reduces harder-to-debug errors/issues deeper into the call stack
- it's consistent with use of introspection queries elsewhere in the library
- it's consistent with other `internal.Api` calls in the `Run` class (I think 4-5 currently)

Reply:
(ignore last comment -- added introspection query into core logic)